src/GoodsListElement/GoodsListElement.js

```diff
@@ -1,22 +1,107 @@
 import React, { Component } from 'react'
 import './GoodsListElement.css'
+import CategoryGoods from '../CategoryGoods/CategoryGoods'
+import { categories } from '../Mocks/CategotyGoods'
+import PropTypes from 'prop-types'
 
 export default class GoodsListElement extends Component {
+    state = {
+        title: '',
+        weight: '',
+        description: '',
+        category: 'Any product',
+        edit: false
+    }
+
     onDelete = (e) => {
         this.props.onDelete(this.props.good.id)
     }
 
+    onToggle = (e) => {
+        this.props.onToggle(this.props.good.id)
+    }
+
+    onInputChange = ({ target }) => {
+        this.setState({
+            [target.name]: target.value
+        })
+    }
+
+    onEdit = (e) => {
+        e.stopPropagation()
+        const {good: {title, weight, description, category} } = this.props;
+        this.setState({
+          title,
+          weight,
+          description,
+          category,
+          edit: true
+        })
+      }
+
+    onSave = () => {
+        const { title, weight, description, category } = this.state
+        const { good: {id}, onSave } = this.props
+        this.setState({
+            edit: false
+        })
+        onSave(id, { title, weight, description, category })
+    }
+
     render() {
-        const { title, weight, description } = this.props.good
+        const { title, weight, description, category} = this.props.good
         return (
             <div className="GoodsListElement">
-                <div className="GoodsListElement_Column">{title}</div>
-                <div className="GoodsListElement_Column">{weight}</div>
-                <div className="GoodsListElement_Column GoodsListElement_ColumnDescription">{description}</div>
+                <div className="GoodsListElement_Column GoodsListElement_Checkbox">
+                    <input 
+                            type='checkbox'
+                            onChange={() => this.onToggle()}/>
+                </div>
```

Comment:
checkbox value should be set from the state. Don't rely on current state that you don't control.